app/stac_api/models.py

```diff
@@ -229,6 +231,52 @@ def update_extent(self, item_properties_datetime):
         #     raise ValidationError(_("Updating the collection extent due to item update failed."))
         pass  # pylint: disable=unnecessary-pass
 
+    def update_bbox_extent(self, action, item_geom, item_id):
+        '''
+        updates the collection's spatial extent when an item is updated.
+        :param action: eighter up (insert, update) or rm (delete)
+        :param geometry: the geometry of the item
+        :param item_id the id of the item being treated
+        This function generates a new extent regarding all the items with the same
+        collection foreign key. If there is no spatial bbox yet, the one of the geometry of the
+        item is being used.
+        '''
+
+        # insert (as item_id is None)
+        if action == 'up' and item_id is None:
+            # the first item of this collection
+            if self.extent_geometry is None:
+                self.extent_geometry = Polygon.from_bbox(GEOSGeometry(item_geom).extent)
+            # there is already a geometry in the collection a union of the geometries
+            else:
+                self.extent_geometry = Polygon.from_bbox(
+                    GEOSGeometry(self.extent_geometry).union(GEOSGeometry(item_geom)).extent)
+
+        # update
+        if action == 'up' and item_id:
+            # is the new bbox larger than (and covering) the existing
+            if Polygon.from_bbox(GEOSGeometry(item_geom).extent).covers(self.extent_geometry):
+                self.extent_geometry = Polygon.from_bbox(GEOSGeometry(item_geom).extent)
+            # we need to iterat trough the items
+            else:
+                qs = Item.objects.filter(collection_id=self.pk).exclude(id=item_id)
+                union_geometry = GEOSGeometry(item_geom)
+                for item in qs:
+                    union_geometry = union_geometry.union(item.geometry)
+                self.extent_geometry = Polygon.from_bbox(union_geometry.extent)
+
+        # delete, we need to iterate trough the items
+        if action == 'rm':
+            qs = Item.objects.filter(collection_id=self.pk).exclude(id=item_id)
+            union_geometry = GEOSGeometry('Multipolygon EMPTY')
+            for item in qs:
+                union_geometry = union_geometry.union(item.geometry)
+            self.extent_geometry = Polygon.from_bbox(union_geometry.extent)
```

Comment:
We can leave this case like this for the moment. We'll have to change this to an asynchronous, e.g. daily cleanup task. Imho this extent information on the collection is kind of a 'nice to have' info for a client system, (we should in any case aks inser if they will rely on it to be 'realtime' accurat). If not, we can add this to the spec saying that we update this info e.g. once a day.